src/main/scala/io/iohk/ethereum/db/storage/ReferenceCountNodeStorage.scala

```diff
@@ -166,15 +166,21 @@ object ReferenceCountNodeStorage extends PruneSupport with Logger {
     * @param nodeStorage
     * @return
     */
-  private def getNodesToBeRemovedInPruning(blockNumber: BigInt, snapshotKeys: Seq[NodeHash], nodeStorage: NodeStorage): Seq[NodeHash] =
-    snapshotKeys.foldLeft(Seq.empty[Option[NodeHash]]) { (nodesToRemove, snapshotKey) =>
-      val toRemove = for {
-        snapshot <- nodeStorage.get(snapshotKey).map(snapshotFromBytes)
+  private def getNodesToBeRemovedInPruning(blockNumber: BigInt, snapshotKeys: Seq[NodeHash], nodeStorage: NodeStorage): Seq[NodeHash] = {
```

Comment:
Do you think maybe a similar improvement in [`update`](https://github.com/input-output-hk/mantis/blob/phase/release1_1/src/main/scala/io/iohk/ethereum/db/storage/ReferenceCountNodeStorage.scala#L52) also could improve the performance?

Reply:
probably yes, but these are nodes from one transaction (there is relativly few of them) so i never saw it bottlenecked in that place, so for now i would leave it at leat until further works of improving pruning performance